sunpy/net/fido_factory.py

```diff
@@ -347,9 +347,9 @@ def fetch(self, *query_results, path=None, max_conn=5, progress=True,
         >>> filepaths = Fido.fetch(filepaths)  # doctest: +SKIP
 
         """
-        if path is not None and not os.access(path, os.W_OK):
+        if path is not None and not os.access(str(path).split("{")[0], os.W_OK):
```

Comment:
There has to be a better way than this. This was all I could come up with.

Reply:
I think you want to take [`path.parts`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parts), and just use all the parts that come before the first part that contains a `{`.

Reply:
more than that, you want to check the highest folder that actually exists, because the downloader will create folders in the tree if they don't

Reply:
the best way I could figure out was this:
```python
>>> path = Path("/home/stuart/does/not/exist")                                                                                                                                                                                     
>>> exists = list(filter(lambda p: p.exists(), path.parents))                                                                                                                                                                      
>>> os.access(exists[0], os.W_OK)                                                                                                                                                                                                  
True
```

Reply:
Copied and pasted into the code.